lib/rucio/api/dirac.py

```diff
@@ -40,31 +42,33 @@ def add_files(lfns, issuer, ignore_availability):
     :param lfns: List of lfn (dictionary {'lfn': <lfn>, 'rse': <rse>, 'bytes': <bytes>, 'adler32': <adler32>, 'guid': <guid>, 'pfn': <pfn>}
     :param issuer: The issuer account.
     :param ignore_availability: A boolean to ignore blocked sites.
+    :param vo: The VO to act on.
+
     """
-    scopes = list_scopes()
+    scopes = list_scopes(vo=vo)
     dids = []
     rses = {}
     for lfn in lfns:
         scope, name = extract_scope(lfn['lfn'], scopes)
         dids.append({'scope': scope, 'name': name})
         rse = lfn['rse']
         if rse not in rses:
-            rse_id = get_rse_id(rse=rse)
+            rse_id = get_rse_id(rse=rse, vo=vo)
             rses[rse] = rse_id
         lfn['rse_id'] = rses[rse]
 
     # Check if the issuer can add dids and use skip_availabitlity
     for rse in rses:
         rse_id = rses[rse]
         kwargs = {'rse': rse, 'rse_id': rse_id}
-        if not has_permission(issuer=issuer, action='add_replicas', kwargs=kwargs):
-            raise AccessDenied('Account %s can not add file replicas on %s' % (issuer, rse))
-        if not has_permission(issuer=issuer, action='skip_availability_check', kwargs=kwargs):
+        if not has_permission(issuer=issuer, action='add_replicas', kwargs=kwargs, vo=vo):
```

Comment:
I think it would be better to add the `vo` to the `kwargs` (above) instead.

Reply:
Looking at the has_permission, may other places (104 different calls) all call this function and specify the vo separately 
often as - `if not has_permission(issuer=issuer, vo=vo, action='various actions', kwargs=kwargs):`

Is it not then beyond the scope of this PR?

Reply:
You are right, I made a mistake here by directly looking at the core `has_permission` method, which does not have to `vo` parameter. But `api.has_permission` has it, which is the correct one. 👍 
You can ignore my comment, consider it resolved 😄